GW2EIEvtcParser/EncounterLogic/Fractals/SilentSurf/Kanaxai.cs

```diff
@@ -297,6 +301,117 @@ internal override void ComputePlayerCombatReplayActors(AbstractPlayer player, Pa
             }
+
+            // Axe AoE attached to players
+            IEnumerable<Segment> buffs69195 = player.GetBuffStatus(log, RendingStormAxeTargetBuff1, log.FightData.LogStart, log.FightData.LogEnd).Where(x => x.Value > 0);
+            IEnumerable<Segment> buffs69208 = player.GetBuffStatus(log, RendingStormAxeTargetBuff2, log.FightData.LogStart, log.FightData.LogEnd).Where(x => x.Value > 0);
+            var axes = new List<Segment>();
+            axes.AddRange(buffs69208);
+            axes.AddRange(buffs69195);
+            foreach (Segment segment in axes)
+            {
+                replay.Decorations.Add(new CircleDecoration(true, 0, 180, ((int)segment.Start, (int)segment.End), "rgba(200, 120, 0, 0.2)", new AgentConnector(player)));
+                replay.Decorations.Add(new CircleDecoration(true, (int)segment.End, 180, ((int)segment.Start, (int)segment.End), "rgba(200, 120, 0, 0.2)", new AgentConnector(player)));
+            }
+
+            // Numbers spread AoEs
+            IEnumerable<Segment> spreads = player.GetBuffStatus(log, KanaxaiSpreadOrangeAoEBuff, log.FightData.LogStart, log.FightData.LogEnd).Where(x => x.Value > 0);
+            foreach (Segment spreadSegment in spreads)
+            {
+                replay.Decorations.Add(new CircleDecoration(true, 0, 380, ((int)spreadSegment.Start, (int)spreadSegment.End), "rgba(200, 120, 0, 0.2)", new AgentConnector(player)));
+            }
+        }
+
+        internal override void ComputeEnvironmentCombatReplayDecorations(ParsedEvtcLog log)
+        {
+            base.ComputeEnvironmentCombatReplayDecorations(log);
+
+            // Red AoE during Frightening Speed attack
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.FrighteningSpeedRedAoE, out IReadOnlyList<EffectEvent> frighteningSpeedRedAoEs))
+            {
+                foreach (EffectEvent aoe in frighteningSpeedRedAoEs)
+                {
+                    int duration = 1500;
+                    int start = (int)aoe.Time;
+                    int effectEnd = start + duration;
+                    EnvironmentDecorations.Add(new CircleDecoration(true, 0, 380, (start, effectEnd), "rgba(255, 0, 0, 0.2)", new PositionConnector(aoe.Position)));
+                    EnvironmentDecorations.Add(new CircleDecoration(false, 0, 380, (start, effectEnd), "rgba(255, 0, 0, 0.2)", new PositionConnector(aoe.Position), 10));
+                }
+            }
+
+            // Axe AoE
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.AxeGroundAoE, out IReadOnlyList<EffectEvent> axeAoEs))
+            {
+                foreach (EffectEvent aoe in axeAoEs)
+                {
+                    // Get World Cleaver casts and find the first cast time event present after the AoE effect time
+                    AgentItem agentKanaxai = log.AgentData.GetAgentByType(AgentItem.AgentType.NPC).FirstOrDefault(x => x.ID == (int)ArcDPSEnums.TargetID.KanaxaiScytheOfHouseAurkusCM);
```

Comment:
Targets.FirstOrDefault(x => x.IsSpecies(TargetID.KanaxaiScytheOfHouseAurkusCM));
No need to create another Kanaxai NPC x)
You have access to Targets, ThrashMobs in here + log.PlayerList and log.Friendlies.
For future reference, if you are looking for an AbstractSingleActor but only have access to AgentItem, you have the log.FindActor API that'll properly create one for you if necessary (and cache it for future uses) or return an already existing one.